src/components/overlays/CompanyCertificateDetails/index.tsx

```diff
@@ -61,22 +64,30 @@ export default function CompanyCertificateDetails({
   const close = () => {
     dispatch(closeOverlay())
   }
+  const [getDocument] = useFetchDocumentMutation()
+  const { data: companyDetails, isFetching } = useFetchOwnCompanyDetailsQuery()
+  const { data } = useFetchCompanyCertificateQuery(companyDetails?.bpn ?? '')
 
-  const { data } = useFetchCertificatesQuery()
-  const { data: document } = useFetchDocumentQuery(id ?? '')
-
-  const companyData = data?.content.filter((cert) => cert.documentId === id)
+  const companyData = data?.filter((cert) => cert.documentId === id)
   const [selected, setSelected] = useState<ComapnyCertificateData>()
   const [pdf, setPdf] = useState<string>()
 
-  if (companyData && companyData?.length > 0) {
-    setSelected(companyData[0])
+  const getDoc = async () => {
+    const response = await getDocument(id).unwrap()
+    const file = response.data
+    const fileURL = URL.createObjectURL(file)
+    setPdf(fileURL)
   }
 
   useEffect(() => {
-    //set file here
-    setPdf('')
-  }, [document])
+    getDoc()
+  }, [])
```

Comment:
we don't need this useEffect if getDocument is implemented as query and not a mutation - which is the right way here